core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -667,6 +700,56 @@ public void accept(Path path) {
         return new JarBuildItem(initJar, null, libDir, packageConfig.type, null);
     }
 
+    private void downloadFernflowerJar(PackageConfig packageConfig, Path fernflowerJar) {
+        String downloadURL = String.format("https://jitpack.io/com/github/fesh0r/fernflower/%s/fernflower-%s.jar",
+                packageConfig.fernflower.hash, packageConfig.fernflower.hash);
+        try (BufferedInputStream in = new BufferedInputStream(new URL(downloadURL).openStream());
+                FileOutputStream fileOutputStream = new FileOutputStream(fernflowerJar.toFile())) {
+            byte[] dataBuffer = new byte[1024];
+            int bytesRead;
+            while ((bytesRead = in.read(dataBuffer, 0, 1024)) != -1) {
+                fileOutputStream.write(dataBuffer, 0, bytesRead);
+            }
+        } catch (IOException e) {
+            log.debug("Unable to download Fernflower from " + downloadURL, e);
```

Comment:
Given you explicitly enable it, I think this should be logged as an error.

Reply:
You are probably right. I was on the fence, but I'll go with your suggestion

Reply:
Fixed